packages/components/src/dashicon/index.js

```diff
@@ -27,7 +27,7 @@ export default class Dashicon extends Component {
 	}
```

Comment:
Immediately above this is a `shouldComponentUpdate`, which will prevent the component from being re-rendered even if we pass a new / different set of `extraProps`.

Reply:
Right. Thank you for pointing that out!

Why are we doing this check with each prop? 
Is there any special prop that we don't want it to make the component update?

Reply:
Originally there were very few props, probably not more than just `icon` and `size`, and it was considered mostly a static component which would never need to be re-rendered.

Reply:
Thanks for the explanation!
I'm still not sure what's the best approach to follow.
Currently I believe that we are re-rendering icons in some cases.

Would you recommend to remove those checks and let React do its job?
Or is there a good way of comparing `extraProps` directly?

Reply:
Something like [`@wordpress/is-shallow-equal`](https://github.com/WordPress/gutenberg/tree/master/packages/is-shallow-equal) can provide some easier mechanism to compare all props:

```js
shouldComponentUpdate( props, nextProps ) {
	return ! isShallowEqual( props, nextProps );
}
```

Which is effectively the same as using [`pure` from `@wordpress/compose`](https://github.com/WordPress/gutenberg/tree/master/packages/compose#pure).

But it's lost much of its value here. I don't know that it'd be necessary at all.